pandas-stubs/io/formats/style.pyi

```diff
@@ -1,97 +1,267 @@
+        cell_ids: bool = ...,
         na_rep: str | None = ...,
-    ): ...
+        uuid_len: int = ...,
+        decimal: str | None = ...,
+        thousands: str | None = ...,
+        escape: str | None = ...,
+        formatter: ExtFormatter | None = ...,
+    ) -> None: ...
+    # def render(self,sparse_index: bool | None = ...,sparse_columns: bool | None = ...,**kwargs,) -> str: ...
+    def set_tooltips(
+        self,
+        ttips: DataFrame,
+        props: CSSProperties | None = ...,
+        css_class: str | None = ...,
+    ) -> Styler: ...
     def to_excel(
         self,
         excel_writer,
         sheet_name: str = ...,
         na_rep: str = ...,
-        float_format=...,
-        columns=...,
-        header: bool = ...,
+        float_format: str | None = ...,
+        columns: Sequence[Hashable] | None = ...,
+        header: Sequence[Hashable] | bool = ...,
         index: bool = ...,
-        index_label=...,
+        index_label: IndexLabel | None = ...,
         startrow: int = ...,
         startcol: int = ...,
-        engine=...,
+        engine: str | None = ...,
         merge_cells: bool = ...,
-        encoding=...,
+        encoding: str | None = ...,
         inf_rep: str = ...,
         verbose: bool = ...,
-        freeze_panes=...,
+        freeze_panes: tuple[int, int] | None = ...,
     ) -> None: ...
-    def format(self, formatter, subset=..., na_rep: str | None = ...): ...
-    def render(self, **kwargs): ...
-    def __copy__(self): ...
-    def __deepcopy__(self, memo): ...
+    def to_latex(
+        self,
+        buf: FilePath | WriteBuffer[str] | None = ...,
+        *,
+        column_format: str | None = ...,
+        position: str | None = ...,
+        position_float: str | None = ...,
+        hrules: bool | None = ...,
+        clines: str | None = ...,
+        label: str | None = ...,
+        caption: str | tuple | None = ...,
+        sparse_index: bool | None = ...,
+        sparse_columns: bool | None = ...,
+        multirow_align: str | None = ...,
+        multicol_align: str | None = ...,
+        siunitx: bool = ...,
+        environment: str | None = ...,
+        encoding: str | None = ...,
+        convert_css: bool = ...,
+    ): ...
+    def to_html(
+        self,
+        buf: FilePath | WriteBuffer[str] | None = ...,
+        *,
+        table_uuid: str | None = ...,
+        table_attributes: str | None = ...,
+        sparse_index: bool | None = ...,
+        sparse_columns: bool | None = ...,
+        bold_headers: bool = ...,
+        caption: str | None = ...,
+        max_rows: int | None = ...,
+        max_columns: int | None = ...,
+        encoding: str | None = ...,
+        doctype_html: bool = ...,
+        exclude_styles: bool = ...,
+        **kwargs,
+    ): ...
+    def set_td_classes(self, classes: DataFrame) -> Styler: ...
+    def __copy__(self) -> Styler: ...
+    def __deepcopy__(self, memo) -> Styler: ...
```

Comment:
Do we need to type `__copy__()` and `__deepcopy__()` ?  Don't they return the same types by default?

Reply:
I tend to leave magic methods in just in case.